### ISSUE TITLE
Can't tap on widget previously found by `scrollTo()`

### ISSUE THREAD

**Issue by jBorkowska** [leancodepl/patrol#1046]
Code can be found here: https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/integration_test/test_test.dart#L25
In this line, we scroll to a widget and then try to tap on it. Scrolling works fine, but tap throws an error:

![Image](https://user-images.githubusercontent.com/70332130/223160131-5d6d7d0a-eda3-4a92-a1c8-cedac645d16f.png)

Possible reason: there are 2 widgets with the same key in the widget tree at this moment in test: a container in https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/lib/ui/screens/home/wonders_home_screen.dart#L293 and a stack in https://github.com/jBorkowska/flutter-wonderous-app/blob/patrol-bug/lib/ui/wonder_illustrations/common/animated_clouds.dart#L100. The container is found by scrollTo, but in error message from tap only the stack is mentioned (as offscreen). Also stack is higher in the widget tree than container, which means it's analyzed before container when resolving the finder. That potentially means, that scrollTo analyzed stack, but ignored it as offstage, but tap took first one from the tree and then checked hitTestable(), which returned false, resulting in exception from screenshot above.

Fix (to be reviewed by @bartekpacia ):
tldr: `scrollTo()` has to pass original finder or `finder.hitTestable().first`.
`scrollTo()` returns `finder.first` instead of original finder. So in this case it returned the stack widget, which is not hittestable. This was passed to `tap()` method and that is why the error message printed only this stack widget (`tap()` called `hitTestable()` on finder passed from `scrollTo()`, that's why it threw an error).

Another issue found in the process: https://github.com/leancodepl/patrol/issues/1048.

**Comment by bartekpacia** [comment]
I can reproduce this.

**Passes**


```dart
void main() {
  patrolTest('can tap() on widget scrolled to by scrollTo()', ($) async {
    await $.pumpWidgetAndSettle(
      MaterialApp(
        home: Scaffold(
          body: ListView(
            children: [
              Stack(
                children: [
                  const Center(child: Text('Text')),
                  Center(
                    child: Container(
                      width: 100,
                      height: 100,
                      color: Colors.blue,
                    ),
                  ),
                ],
              ),
              const Text('Text'),
            ],
          ),
        ),
      ),
    );

    await $('Text').scrollTo();
    await $('Text').tap();
  });
}
```

**Fails**

```dart
void main() {
  patrolTest('can tap() on widget scrolled to by scrollTo()', ($) async {
    await $.pumpWidgetAndSettle(
      MaterialApp(
        home: Scaffold(
          body: ListView(
            children: [
              Stack(
                children: [
                  const Center(child: Text('Text')),
                  Center(
                    child: Container(
                      width: 100,
                      height: 100,
                      color: Colors.blue,
                    ),
                  ),
                ],
              ),
              const Text('Text'),
            ],
          ),
        ),
      ),
    );

    await $('Text').scrollTo().tap();
  });
}

```
